metals/src/main/scala/scala/meta/internal/metals/codeactions/CreateNewSymbol.scala

```diff
@@ -37,15 +37,40 @@ class CreateNewSymbol() extends CodeAction {
       )
     }
 
+    def createNewMethod(
+        diagnostic: l.Diagnostic,
+        name: String,
+    ): l.CodeAction = {
+      val codeAction = new l.CodeAction()
+      codeAction.setTitle(CreateNewSymbol.method(name))
+      codeAction.setKind(l.CodeActionKind.QuickFix)
+      codeAction.setDiagnostics(List(diagnostic).asJava)
+      codeAction.setCommand(
+        ServerCommands.InsertInferredMethod.toLsp(
+          new l.TextDocumentPositionParams(
+            params.getTextDocument(),
+            params.getRange().getStart(),
+          )
+        )
+      )
+      codeAction
+    }
+
     val codeActions = params
       .getContext()
       .getDiagnostics()
       .asScala
       .groupBy {
-        case ScalacDiagnostic.SymbolNotFound(name) => Some(name)
+        case ScalacDiagnostic.SymbolNotFound(name) if name.nonEmpty =>
+          Some(name)
         case _ => None
       }
       .collect {
+        case (Some(name), diags)
+            if name.head.isLower && params
+              .getRange()
+              .overlapsWith(diags.head.getRange()) =>
+          createNewMethod(diags.head, name)
```

Comment:
Let's also check what version of Scala it is since Scala 3is not implemented

Reply:
done in https://github.com/scalameta/metals/pull/6877/commits/12ec7f86a0245b74b3a1b5597d466250bcac58ec